src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -621,15 +624,18 @@ public TypeSymbolWithAnnotations SetUnknownNullabilityForReferenceTypes()
         private class WithoutCustomModifiers : TypeSymbolWithAnnotations
         {
             protected readonly TypeSymbol _typeSymbol;
+            protected readonly bool? _isNullableIfReferenceType;
```

Comment:
>protected readonly bool? _isNullableIfReferenceType; [](start = 12, length = 52)

This feels wrong, this value is not delayed, we should be creating the right subclass instead. It looks like there is a Create factory that does that ```public static TypeSymbolWithAnnotations Create(TypeSymbol typeSymbol, bool? isNullableIfReferenceType)```

Reply:
Modifiers can be added via WithModifiers helper.

---
In reply to: [171115378](https://github.com/dotnet/roslyn/pull/25087#discussion_r171115378) [](ancestors = 171115378)